src/components/Catalog.js

```diff
@@ -1,36 +1,112 @@
-import React from 'react';
+import React, {Component} from 'react';
 import {NavLink} from 'react-router-dom';
-
 import '../styles/Catalog.css';
 
-const Catalog = props => {
-  const {products} = props;
+import {connect} from 'react-redux';
+import {fetchCategories} from '../redux-store/actions/category-actions';
+import {
+  fetchProducts,
+  fetchProductsByCategory,
+  searchForProductsInDatabase,
+} from '../redux-store/actions/product-actions';
+
+class Catalog extends Component {
+  constructor() {
+    super();
+    this.state = {
+      categories: [],
+      products: [],
+    };
+  }
+
+  setLocalProps = () => {
+    const {location, match} = this.props;
+    const urlArray = location.pathname.split('/');
+
+    const filterKeyword = urlArray[urlArray.indexOf('catalog') + 1];
+
+    this.props.fetchCategories();
```

Comment:
I still think app should fetch initial state for redux store since it's the first thing that loads, if I changed the Home page to not render catalog like we discussed, we would never have this initial fetch